rewrite-test/src/main/kotlin/org/openrewrite/java/typegeneration/TypeGenerationTest.kt

```diff
@@ -0,0 +1,512 @@
+            }
+        )
+    }
+
+    //Need to handle generics better before we can enable this test.
+    @Test
+    @Disabled
+    fun testFunctionalInterfaceWithLambdas() {
+        assertCompilationWithByteCode(
+            parser = JavaParser.fromJavaVersion()
+                .classpath("")
+                .dependsOn(Arrays.asList(
+                    Parser.Input.fromString("""
+                        package org.example.interfaces;
+                        import java.util.*;
+                        
+                        public interface NoGenericsNoReturn {
+                            public void yodel(String lay, long he, Integer hoo);  
+                        }
+                        public interface NoGenerics {
+                            public Date yodel(String lay, long he, Integer hoo);  
+                        }
+
+                        public interface Generics<T, R> {
+                            public R yodel(T lay, List<String> he, Set<Integer> hoo);  
+                        }
+                        
+                        public interface YodelUtils {
+                            public void yodelSome(NoGenericsNoReturn song);
+                            public void yodelMore(NoGenerics song);
+                            public void tooMuchYodelling(Generics<String, String>  song);
+                        }
+                        """))
+                ).build(),
+            source = """
+                    package org.example;
+                    
+                    import org.example.interfaces.*;
+                    import java.util.*;
+                    
+                    public class Example {
+                        YodelUtils yodellingUtils;
+                        public void yodeller() {
+                            yodellingUtils.yodelSome((l, h, hoo) -> {});
+                            yodellingUtils.yodelMore((l, h, hoo) -> new Date());
+                            yodellingUtils.tooMuchYodelling((l, h, hoo) -> "hello");
+                        }
+                    }
+            """.trimIndent(),
+            checkCompilerOutput = System.out::println,
+            checkCompilationUnit = { compilationUnit ->
+
+                val method = compilationUnit.classes[0].body.statements[1] as J.MethodDeclaration
+
+                val yodelSome = method.body!!.statements[0] as J.MethodInvocation
+                val yodelMore = method.body!!.statements[1] as J.MethodInvocation
+                val tooMuchYodelling = method.body!!.statements[2] as J.MethodInvocation
+                assertThat((yodelSome.arguments[0] as J.Lambda).type).isNotNull
+                assertThat((yodelMore.arguments[0] as J.Lambda).type).isNotNull
+                assertThat((tooMuchYodelling.arguments[0] as J.Lambda).type).isNotNull
+            }
+        )
+    }
+
+    //Because the structure of these source files can (and likely will) change, these should only be run manually.
```

Comment:
Realistically manual tests wont ever be run. 
You could stuff testing-only sources into src/test/resources and load them like any other resource

Reply:
good idea here.